pkg/proxy/docker/docker.go

```diff
@@ -469,9 +469,21 @@ func (d *ContainerTruststorePatcher) leasePatchSet(container string) *patchSet {
 
 // Proxy serves the Docker API while patching the container truststore.
 func (d *ContainerTruststorePatcher) Proxy(srvAddr, dockerAddr string) {
-	tcpChan := make(chan net.Conn, 1)
+	dockerProxyChan := make(chan net.Conn, 1)
 	udsChan := make(chan net.Conn, 1)
-	tl, err := net.Listen("tcp", srvAddr)
+
+	network := "tcp"
+	address := srvAddr
+	if strings.HasPrefix(address, "unix://") {
+		network = "unix"
+		address = strings.TrimPrefix(address, "unix://")
+		if _, err := os.Stat(address); err == nil {
+			if err := os.RemoveAll(address); err != nil {
```

Comment:
woah... I think maybe we should fail instead in this case. it seems like net.Listen should unlink the socket on exit so this would only be an issue if there were an outstanding proxy (in which case we'd break the other proxy by removing it) or another process on the system created/is using it (in which case we'd really want the caller to clean it up). Best just to leave it